### ISSUE TITLE
Dynamic update of tile

### ISSUE THREAD

**Issue by G9YH** [recastnavigation/recastnavigation#342]
Hello,

    I wonder if I could divide meshes of scene into tiles, just like the navmesh does, in this way once I update my scene, like adding an abstacle in point A, only the tile A belongs to needs to update.
    
Thanks!!

**Comment by jakobbotsch** [comment]
Sorry, I don't quite understand your question. As you mention, Recast already supports tiles. Could you please try to provide some more details?

**Comment by nkari82** [comment]
Is there a way to get the associated tile id when the navmesh is reconstructed due to an obstacle?

**Comment by jakobbotsch** [comment]
No, but it is not hard to query the tiles that will be affected by using `getObstacleBounds` and `queryTiles`. See [dtTileCache::update](https://github.com/recastnavigation/recastnavigation/blob/05b2b8da80037887d3e79af8a59a8f0f8ed02602/DetourTileCache/Source/DetourTileCache.cpp#L538-L555) for how the tile cache determines which tiles to reconstruct when an obstacle is added using these methods.

**Comment by G9YH** [comment]
Sorry, I may not make it clear. I see that dtTileCache::addObstacle can add a cylinder obstacle, but is there a way to add an irregular obstacle?

**Comment by jakobbotsch** [comment]
Unfortunately the tile cache supports only boxes (axis-aligned and oriented) or cylinders. There is no technical reason for this, that is just what's implemented currently.

**Comment by G9YH** [comment]
Thanks a lot! In this way, it can not add irregular obstacles into scene if do not reload all meshes by using InputGeom::load, right?

**Comment by G9YH** [comment]
Thanks for your time, I'll try this!
